lib/hanikamu/service.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "dry-struct"
+require "dry-monads"
+
+module Hanikamu
+  # :nodoc
+  class Service < Dry::Struct
+    include Dry::Monads[:result]
+    WhiteListedError = Class.new(StandardError)
```

Comment:
If I want to use this outside this, I think it would make sense to declare this somewhere else.

So for example when I am declaring a guard error in my project, I can inherit from

Hanikamu::ServiceError

Reply:
I think WhitelistedError is a bad naming. I would prefer Hanikamu::ServiceError

Reply:
What about :
Hanikamu::Service::Error
So we can keep the same structure?